### PR TITLE
dbus cleanup

### PR DESCRIPTION
**Pull request by Meumeu**
This PR includes a few things that are needed to be able to experiment using rog-core from python (or maybe other languages):
- Add the org.freedesktop.DBus.Introspectable interface
- Set different names for the arguments (or else python raises an exception when connecting)
- Add method_return()s where missing

### CODE REVIEW

**Comment by flukejones** [comment]
Ah cool, thanks for this! I'm a bit new to dbus so any help I can get here is appreciated.

**Comment by flukejones** [comment]
Since you are using python, would having a python wrapper for the client lib be of any use to you?

File: rog-core/src/rog_dbus.rs [hunk 1]
```diff
@@ -78,23 +78,23 @@ pub(super) fn dbus_create_ledeffect_method(effect: NestedVecType) -> Method<MTSy
                         iter.read()?,
                     ];
                     *lock = Some(byte_array);
-                    Ok(vec![])
+                    Ok(vec!(m.msg.method_return()))
```
[the remark this answers is not in the document]
Comment (Meumeu) [reply rog-core/src/rog_dbus.rs:81]: I haven't used dbus a lot either, but any other client will just wait for a reply. OTOH I've just discovered that you can set the `org.freedesktop.DBus.Method.NoReply` annotation to `true` when you create the method to tell the client it shouldn't expect a reply, that's probably cleaner.

**Comment by Meumeu** [comment]
> Since you are using python, would having a python wrapper for the client lib be of any use to you?

I've mostly used python for some quick testing, the real stuff I'm doing is in C++ actually, and I'm using a dbus proxy that is automatically generated from the dbus introspection data. So that I won't really need it.

File: rog-core/src/rog_dbus.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (flukejones) [reply rog-core/src/rog_dbus.rs:81]: I would prefer that NoReply is used instead. The reply is meaningless in pretty much all cases here (even with the other dbus methods) as there is no real info attached beyond ack. The actual device writes happen async to this.

**Comment by flukejones** [comment]
> > Since you are using python, would having a python wrapper for the client lib be of any use to you?
> 
> I've mostly used python for some quick testing, the real stuff I'm doing is in C++ actually, and I'm using a dbus proxy that is automatically generated from the dbus introspection data. So that I won't really need it.

The client stuff here I'm referring to is the keyboard RGB layouts as detailed [here](https://github.com/flukejones/rog-core/blob/master/rog-client/src/fancy.rs), since you are playing with the RGB effects I thought you might be interested in that. The actual dbus stuff is more convenience than anything - I was pretty close to just using a unix socket for the client/server communications :) Probably would have been cleaner and easier.

**Pull request merged**